### PR TITLE
Initial support of breakpoint setting by source file + line on Linux using portable pdb reader

### PR DESCRIPTION
**Pull request by lucenticus**
This pull request enables `bpmd source:line` SOS command on Linux platform using portable pdb reader from pull request https://github.com/dotnet/corefx/pull/9281.
We have to put `System.Diagnostics.Debug.SymbolReader.dll` to the directory with debuggee application.
Also, we have to build application with portable pdb using `/debug:portable` option and put pdb file to the directory with debuggee application.
This feature was tested on Ubuntu 14.04 Linux x64.
Related issue: #4798
CC: @mikem8361, @janvorli, @Dmitri-Botcharnikov, @seanshpark, @chunseoklee, @kvochko, @sam-baravy

### CODE REVIEW

**Comment by mikem8361** [comment]
You should fix the build errors before we review.

File: src/ToolBox/SOS/Strike/strike.cpp [hunk 1]
```diff
@@ -7110,9 +7117,13 @@ DECLARE_API(bpmd)
         // did we get dll and type name or file:line#? Search for a colon in the first arg
         // to see if it is in fact a file:line#
         CHAR* pColon = strchr(DllName.data, ':');
+        if (!SUCCEEDED(g_ExtSymbols->GetModuleByModuleName(MAIN_CLR_DLL_NAME_A, 0, NULL, NULL))) {
```
Comment (mikem8361) [inline src/ToolBox/SOS/Strike/strike.cpp]: You should use FAILED here instead of !SUCCEEDED

**Comment by mikem8361** [comment]
LGTM.

**Comment by lucenticus** [comment]
@mikem8361 Thanks for review

**Comment by Dmitri-Botcharnikov** [comment]
LGTM

**Comment by rahku** [comment]
SOS for windows x64 seems to have stopped working after this change.
In particular !bpmd command no longer works. It always prints the following message.
0:000> !bpmd a.exe Program.Main
File name:Line number not supported
Usage: !bpmd -md <MethodDesc pointer>
Usage: !bpmd [-nofuturemodule] <module name> <managed function name> [<il offset>]
Usage: !bpmd <filename>:<line number>
Usage: !bpmd -list
Usage: !bpmd -clear <pending breakpoint number>
Usage: !bpmd -clearall
See "!help bpmd" for more details.

This PR added this code fragment due to which we always go inside the if block and set fBadParam to true.
        if (FAILED(g_ExtSymbols->GetModuleByModuleName(MAIN_CLR_DLL_NAME_A, 0, NULL, NULL))) {
            ExtOut("File name:Line number not supported\n");
           fBadParam = true;
        }

**Comment by mikem8361** [comment]
Is coreclr loaded? "libcoreclr.so" on Linux.

**Comment by rahku** [comment]
yes it is.

**Comment by rahku** [comment]
It seems we are calling g_ExtSymbols->GetModuleByModuleName with module name as coreclr.dll Otherplaces where the api is called it does not append .dll to the name such as here
https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/Strike/dllsext.cpp#L72

**Comment by mikem8361** [comment]
Not sure why it is failing now because other than the !SUCCEEDED to FAILED change that code has been that way a long time. I also just tried it with the latest and !bpmd seems to work for me.

**Comment by rahku** [comment]
https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/Strike/strike.cpp#L7119
This was added in this PR ...it has not been present before. Changing MAIN_CLR_DLL_NAME_A to MAIN_CLR_MODULE_NAME_A fixes the problem on windows x64. Do you know where is the definition for MAIN_CLR_DLL_NAME_A for linux?

**Comment by rahku** [comment]
Did you try latest on windows x64?

**Comment by mikem8361** [comment]
Yes.

**Comment by mikem8361** [comment]
Sorry, you are right that this was added by the PR. I didn't look back far enough in the history.

MAIN_CLR_MODULE_NAME_* for Linux/OSX is a little tricky because the LLDB version of GetModuleByModuleName expects the DLL name that is the module name with the prefix ("lib") and suffix (".so" or ".dylib").  So the simplest fix is to #ifdef FEATURE_PAL name passed to GetModuleByModuleName.  MAIN_CLR_MODULE_NAME_A for Windows and MAIN_CLR_DLL_NAME for Linux/OS X.

**Comment by mikem8361** [comment]
And yes I can repro this problem with my binaries.  I used "!bpmd -list" to test it instead of what you put above "!bpmd a.exe Program.Main".

**Comment by rahku** [comment]
ok will send PR for fix.

**Pull request merged**